datumaro/plugins/transforms.py

```diff
@@ -541,4 +541,19 @@ def transform_item(self, item):
                     annotations.append(ann.wrap(label=conv_label))
             else:
                 annotations.append(ann.wrap())
+        return item.wrap(annotations=annotations)
+
+class TransformLabels(Transform, CliPlugin):
+    """
+    Collects all labels from annotations (of all types) and
+    transforms them into a set of annotations of type Label
+    """
+
+    def transform_item(self, item):
+        labels = set(p.label for p in item.annotations
```

Comment:
It might be better to keep existing Labels as is (with attributes etc.) and only convert other types. Not all annotation types have labels (example - captions), so it is better to check with `getattr(p, 'label') != None`.